packages/clerk-js/src/ui/elements/CodeControl.tsx

```diff
@@ -157,302 +163,128 @@ export const OTPResendButton = () => {
-      if (values[clampedIndex]) {
-        ref.select();
-      }
-    }
-  };
-
-  const handleOnClick = (index: number) => (e: React.MouseEvent<HTMLInputElement>) => {
-    e.preventDefault();
-    // Focus on the first digit, when the first click happens.
-    // This is helpful especially for mobile (iOS) devices that cannot autofocus
-    // and user needs to manually tap the input area
-    if (!firstClickRef.current) {
-      focusInputAt(0);
-      firstClickRef.current = true;
-      return;
-    }
-    focusInputAt(index);
-  };
-
-  const handleOnChange = (index: number) => (e: React.ChangeEvent<HTMLInputElement>) => {
-    e.preventDefault();
-    handleMultipleCharValue({ eventValue: e.target.value || '', inputPosition: index });
-  };
-
-  const handleOnInput = (index: number) => (e: React.FormEvent<HTMLInputElement>) => {
-    e.preventDefault();
-    if (isValidInput((e.target as any).value)) {
-      // If a user types on an input that already has a value and the new
-      // value is the same as the old one, onChange will not fire so we
-      // manually move focus to the next input
-      focusInputAt(index + 1);
-    }
-  };
-
-  const handleOnPaste = (index: number) => (e: React.ClipboardEvent<HTMLInputElement>) => {
-    e.preventDefault();
-    handleMultipleCharValue({ eventValue: e.clipboardData.getData('text/plain') || '', inputPosition: index });
-  };
-
-  const handleOnKeyDown = (index: number) => (e: React.KeyboardEvent<HTMLInputElement>) => {
-    switch (e.key) {
-      case 'Backspace':
-        e.preventDefault();
-        changeValueAt(index, '');
-        focusInputAt(index - 1);
-        return;
-      case 'ArrowLeft':
-        e.preventDefault();
-        focusInputAt(index - 1);
-        return;
-      case 'ArrowRight':
-        e.preventDefault();
-        focusInputAt(index + 1);
-        return;
-      case ' ':
-        e.preventDefault();
-        return;
-    }
-  };
-
-  // Handle hidden input changes (for password manager autofill)
-  const handleHiddenInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const value = e.target.value.replace(/\D/g, '').slice(0, length);
-    const newValues = value.split('').concat(Array.from({ length: length - value.length }, () => ''));
-    setValues(newValues);
-
-    // Focus the appropriate visible input
-    if (value.length > 0) {
-      focusInputAt(Math.min(value.length - 1, length - 1));
-    }
-  };
-
-  const centerSx = centerAlign ? { justifyContent: 'center', alignItems: 'center' } : {};
-
   return (
     <Box
       elementDescriptor={descriptors.otpCodeFieldInputContainer}
       sx={{ position: 'relative' }}
     >
-      {/* Hidden input for password manager compatibility */}
-      <Input
-        ref={hiddenInputRef}
-        type='text'
-        autoComplete='one-time-code'
-        inputMode='numeric'
-        pattern={`[0-9]{${length}}`}
-        minLength={length}
+      <OTPInput
+        aria-label='Enter verification code'
```

Comment:
Should this be localized?

Reply:
Some of the existing ones weren't. I can follow-up with this.